### PR TITLE
[GEOS-8170] Allow CatalogBuilder to setup a new CoverageInfo even if the test read return null

### PR DESCRIPTION
**Pull request by aaime**

### CODE REVIEW

File: src/main/src/main/java/org/geoserver/catalog/CatalogBuilder.java [hunk 1]
```diff
@@ -1077,13 +1035,88 @@ private CoverageInfo buildCoverageInternal(GridCoverage2DReader reader, String n
...
+            }
+            final int numBands = sm.getNumBands();
+            sampleDimensions = new GridSampleDimension[numBands];
+            // setting bands names.
+            for (int i = 0; i < numBands; i++) {
+                final ColorInterpretation colorInterpretation = TypeMap.getColorInterpretation(cm, i);
+                if (colorInterpretation == null)
+                    throw new IOException("Unrecognized sample dimension type for band number " + (i + 1));
+                sampleDimensions[i] = new GridSampleDimension(colorInterpretation.name());
+            }
```
Comment (dromagnoli) [inline src/main/src/main/java/org/geoserver/catalog/CatalogBuilder.java]: some "if(condition)" bad formatting and an "if(colorInterpretation == null) body " without enclosing bracket.

File: src/main/src/main/java/org/geoserver/catalog/CatalogBuilder.java [hunk 2]
```diff
@@ -1077,13 +1035,88 @@ private CoverageInfo buildCoverageInternal(GridCoverage2DReader reader, String n
...
+        }
+
+        parameters.put(AbstractGridFormat.READ_GRIDGEOMETRY2D.getName().toString(), new GridGeometry2D(testRange, testEnvelope));
+
+        // try to read this coverage
+        final GridCoverage2D gc = reader.read(CoverageUtils.getParameters(readParams, parameters, true));
+        final GridSampleDimension[] sampleDimensions;
+        if (gc != null) {
+            // remove read grid geometry since it is request specific
+            parameters.remove(AbstractGridFormat.READ_GRIDGEOMETRY2D.getName().toString());
```
Comment (dromagnoli) [inline src/main/src/main/java/org/geoserver/catalog/CatalogBuilder.java]: This was already part of the previous code but I'm wondering why it is needed.

Comment (aaime) [inline src/main/src/main/java/org/geoserver/catalog/CatalogBuilder.java]: Indeed, makes no sense, parameters map is built in the method and does not get used outside.

**Comment by aaime** [comment]
@dromagnoli feedback addressed

**Comment by aaime** [comment]
Manually merged

**Pull request closed without merging**